code/game/machinery/camera/camera.dm

```diff
@@ -302,7 +302,7 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/camera/xray, 0)
 	if(!camera_enabled)
 		return
 	. = ..()
-	if(.)
+	if(. && proximity_monitor)
 		triggerCameraAlarm()
 		toggle_cam(null, 0)
```

Comment:
Have you actually tested this change? From the way the code is structured, it looks like the camera might not actually get disabled when destroyed if there isn't a proximity monitor.

Also, if you're making changes to non-modular code, it's best to leave a comment explicitly saying so. Something like `if(. && proximity_monitor) // WALLSTATION EDIT`